packages/integration-tests/src/test-e2e/deployment.test.ts

```diff
@@ -157,28 +157,30 @@ void describe('deployment tests', { concurrency: testConcurrencyLevel }, () => {
           });
 
           void it(`[${testProjectCreator.name}] hot-swaps a change`, async () => {
-            const processController = ampxCli(
-              ['sandbox', '--dirToWatch', 'amplify'],
-              testProject.projectDirPath,
-              {
-                env: sharedSecretsEnv,
-              }
-            );
-
             const updates = await testProject.getUpdates();
-            for (const update of updates) {
-              processController
-                .do(replaceFiles(update.replacements))
-                .do(ensureDeploymentTimeLessThan(update.deployThresholdSec));
-            }
+            if (updates.length > 0) {
```

Comment:
When can the `updates.length` not be greater `0`?

Reply:
Compare https://github.com/aws-amplify/amplify-backend/blob/320a86d053e5131b4c7dc45c7eff29d4e07ba3d6/packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts#L174 
with
https://github.com/aws-amplify/amplify-backend/blob/320a86d053e5131b4c7dc45c7eff29d4e07ba3d6/packages/integration-tests/src/test-project-setup/test_project_base.ts#L122

In other words, only one project has any updates currently. Therefore, for projects without updates this test (in previous form) is 100% overlapping with "deploys fully" above.

Reply:
Also a search https://github.com/search?q=repo%3Aaws-amplify%2Famplify-backend%20%22async%20getUpdates()%22&type=code